### PR TITLE
Implement Barcode API connection

### LINKED ISSUE #11: Scan barcode of book
As a user I want to scan the barcode of a physical book to add this book and its data to my library.

### PR DESCRIPTION
**Pull request by LuisMossburger**
**Description**
Resolves #11.
ISBN barcodes can now be scanned and the respective book (if found) is automatically transferred to the respective shelf.

**Affects**
Barcode Fragment class & layout.

**Notes for Reviewer**
Please review & comment/approve. This tool works quite well:
ISBN that is available through our API: https://barcode.tec-it.com/de/ISBN13?data=9781234567897
ISBN that is not available through our API: https://barcode.tec-it.com/de/ISBN13?data=978-3-030-02111-5

### CODE REVIEW

File: BibBuddy/app/src/main/java/de/bibbuddy/BookBarcodeScannerFragment.java [hunk 1]
```diff
@@ -116,37 +116,83 @@ public void release() {
...
     });
+
+    closeFragment();
+  }
+
+  /**
+   * Closes the BookBarcodeScannerFragment.
+   */
+  public void closeFragment() {
+    LibraryFragment fragment = new LibraryFragment();
```
Comment (claudia3) [inline BibBuddy/app/src/main/java/de/bibbuddy/BookBarcodeScannerFragment.java:179]: Would it be better to open the BookFragment where you see that the new book was added?

File: BibBuddy/app/src/main/java/de/bibbuddy/BookBarcodeScannerFragment.java [hunk 2]
```diff
@@ -116,37 +116,83 @@ public void release() {
       }
 
       /**
-       * Receives the ISBN of a book.
+       * Receives a barcode ISBN, hands the ISBN over the the API.
```
Comment (claudia3) [inline BibBuddy/app/src/main/java/de/bibbuddy/BookBarcodeScannerFragment.java:119]:
```suggestion
       * Receives a barcode ISBN, hands the ISBN over to the API.
```

**Review by claudia3 — COMMENTED** [review]
The book is added to the shelf when it exists in the API. If not a Toast massage appears that the ISBN was not found. The rest seems to work. I suggest the changes that I commented.

**Review by silvia-ivanova-github-rep — CHANGES_REQUESTED** [review]
Please provide comments for the methods (inclusive explanation for the input parameters). 
P.S. when I give an ISBN number in the text field, I am not able to delete/ change the number (in case I give a wrong number).

File: BibBuddy/app/src/main/java/de/bibbuddy/BookBarcodeScannerFragment.java [hunk 3]
```diff
@@ -116,37 +116,83 @@ public void release() {
...
        */
       @Override
       public void receiveDetections(Detector.Detections<Barcode> detections) {
```
[the remark this answers is not in the document]
Comment (LuisMossburger) [reply BibBuddy/app/src/main/java/de/bibbuddy/BookBarcodeScannerFragment.java:122]: As discussed in team, for now we just comment the methods wanted by checkstyle - we will discuss more comments in the team later.

File: BibBuddy/app/src/main/java/de/bibbuddy/BookBarcodeScannerFragment.java [hunk 4]
```diff
@@ -116,37 +116,83 @@ public void release() {
...
-              // TODO check if book exists in API
-              // TODO API get book data
-              // if not exists: display error message and open create new book form alternative
-              // TODO check if valid data / book information data display
-              // TODO add book to shelf
-              // TODO bundle for BookFragment with shelfId and shelfName
-            }
-          });
-
+  private void handleIsbnInput(String isbn) {
```
[the remark this answers is not in the document]
Comment (LuisMossburger) [reply BibBuddy/app/src/main/java/de/bibbuddy/BookBarcodeScannerFragment.java:132]: As discussed in team, for now we just comment the methods wanted by checkstyle - we will discuss more comments in the team later.

File: BibBuddy/app/src/main/java/de/bibbuddy/BookBarcodeScannerFragment.java [hunk 5]
```diff
@@ -116,37 +116,83 @@ public void release() {
...
+      
+      getActivity().runOnUiThread(new Runnable() {
+        public void run() {
+          Toast.makeText(getActivity(), getString(R.string.isbn_not_found),
+              Toast.LENGTH_SHORT).show();
         }
+      });
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (LuisMossburger) [reply BibBuddy/app/src/main/java/de/bibbuddy/BookBarcodeScannerFragment.java:159]: As discussed in team, for now we just comment the methods wanted by checkstyle - we will discuss more comments in the team later.

**Comment by LuisMossburger** [comment]
Deleted instead of hid a comment by @claudia3 where she stated that after a barcode scanning user should return to the shelf instead of the library - for the moment, we think the current implementation is better!

**Review by silvia-ivanova-github-rep — APPROVED** [review]

**Pull request merged**